api/api.proto

```diff
@@ -2163,24 +2171,32 @@ message SdkCloudBackupStatus {
   google.protobuf.Timestamp completed_time = 6;
   // NodeID is the ID of the node where this Op is active
   string node_id = 7;
+  // SourceVolumeID is the the volume that is either being backed up to cloud
+  // or target volume to which cloudsnap is being restored
+  string src_volume_id = 8;
+  // ErrString indicates the failure cause for failed backup/restore
+  string err_string = 9;
 }
 
 // Defines a request to retreive the status of a backup or restore for a
 // specified volume
 message SdkCloudBackupStatusRequest {
   // This is an optional value which is used to get information on the
-  // status of a backup for the specified volume. If no volume id is provided,
-  // then status for all volumes is returned.
+  // status of a backup for the specified volume. If no volume id and TaskId
+  // is provided, then status for all volumes is returned.
   string volume_id = 1;
   // Local indicates if only those backups/restores that are
   // active on current node must be returned
   bool local = 2;
+  // TaskId is the backup/restore task id, if this is specified,
+  // VolumeId is ignored.
+  string task_id = 3;
 }
 
 // Defines a response containing the status of the backups for a specified volume
 message SdkCloudBackupStatusResponse {
   // Statuses is list of currently active/failed/done backup/restores where
-  // the key is the volume ids of current volumes being backed up or restored
+  // the key is the id of the task performing backup/restore.
```

Comment:
Isn't task_id optional? What happens when it not provided, what is the key?

Reply:
Key is the task ID as the status response structure describes.  both task id and volume-id are optional, in which case all existing tasks will be returned. I will add it to the comments.